src/main/java/com/softserve/demo/service/impl/OrderServiceImpl.java

```diff
@@ -91,16 +88,17 @@ public void deleteOrder(Integer id) {
     }
 
     @Override
-    public Resource getResource(String fileName, HttpServletResponse response) {
-        log.info("Get resource name: [{}]", fileName);
+    public Resource getResponseResource(String fileName, HttpServletResponse response) {
+        log.info("Get response resource with name: [{}]", fileName);
         response.setContentType("text/csv; charset=utf-8");
         response.setHeader("Content-Disposition", "attachment; filename=" + fileName);
         response.setHeader("filename", fileName);
-        Resource resource = null;
+        Resource resource;
         try {
-            resource = new UrlResource(new File(pathToContracts + fileName).toPath().toUri());
+            resource = getResource(fileName);
```

Comment:
you can return resource from here without local variable, and return null at the end.